src/send/SendConfirmation.tsx

```diff
@@ -48,39 +48,59 @@ const DEBOUNCE_TIME_MS = 250
 
 export const sendConfirmationScreenNavOptions = noHeader
 
-function useCalculatedFees({
+function useDisplaySendAmounts({
   tokenAmount,
   localAmount,
+  tokenInfo,
   prepareTransactionsResult,
 }: {
+  tokenInfo: TokenBalance | undefined
   tokenAmount: BigNumber | null
   localAmount: BigNumber | null
   prepareTransactionsResult: PreparedTransactionsResult | undefined
 }) {
-  const localCurrencySymbol = useSelector(getLocalCurrencySymbol) ?? LocalCurrencySymbol.USD
   const { maxFeeAmount, feeCurrency: feeTokenInfo } =
     getFeeCurrencyAndAmounts(prepareTransactionsResult)
   const feeAmount = maxFeeAmount ?? new BigNumber(0)
   const localFeeAmount = useTokenToLocalAmount(feeAmount, feeTokenInfo?.tokenId)
 
-  const totalDisplayTokenAmount = useMemo(() => {
-    const total = (tokenAmount ?? new BigNumber(0)).plus(feeAmount ?? new BigNumber(0))
-    return formatValueToDisplay(total)
-  }, [tokenAmount, feeAmount, localCurrencySymbol])
+  const networkFee = useMemo(
+    () => ({
+      localAmount,
+      feeTokenSymbol: feeTokenInfo?.symbol,
+      displayTokenAmount: formatValueToDisplay(feeAmount),
+      displayLocalAmount: localFeeAmount ? formatValueToDisplay(localFeeAmount) : undefined,
+    }),
+    [localFeeAmount, feeAmount, feeTokenInfo]
+  )
 
-  const totalDisplayLocalAmount = useMemo(() => {
-    const total = (localAmount ?? new BigNumber(0)).plus(localFeeAmount ?? new BigNumber(0))
-    return formatValueToDisplay(total)
-  }, [localAmount, localFeeAmount, localCurrencySymbol])
+  const total = useMemo(() => {
+    const totalLocalAmount = localAmount && localFeeAmount ? localAmount.plus(localFeeAmount) : null
+    const displayLocalAmount = totalLocalAmount ? formatValueToDisplay(totalLocalAmount) : ''
 
-  return {
-    localAmount: localFeeAmount,
-    displayTokenAmount: formatValueToDisplay(feeAmount),
-    displayLocalAmount: localFeeAmount ? formatValueToDisplay(localFeeAmount) : undefined,
-    tokenSymbol: feeTokenInfo?.symbol,
-    totalDisplayTokenAmount,
-    totalDisplayLocalAmount,
-  }
+    if (tokenInfo?.tokenId !== feeTokenInfo?.tokenId) {
+      return {
+        type: 'multiple-tokens' as const,
+        token1: {
+          amount: tokenAmount ? formatValueToDisplay(tokenAmount) : '',
+          symbol: tokenInfo?.symbol,
+        },
+        token2: { amount: formatValueToDisplay(feeAmount), symbol: feeTokenInfo?.symbol },
+      }
+    }
+
+    const totalTokenAmount = (tokenAmount ?? new BigNumber(0)).plus(feeAmount ?? new BigNumber(0))
+    const displayTokenAmount = formatValueToDisplay(totalTokenAmount)
+    return {
+      type: 'same-token' as const,
```

Comment:
i would advocate for reserving enum types for concepts that are useful to define - sorry, i don't know how to explain this well. my concern is that this "type" here actually hinders readability by abstracting what a "same token" or "multiple token" type of send is when it's not really necessary for us to define (since in the place that needs to know this, we can just directly compare the send token to the fee token). "same token" and "multiple token" sends are also difficult to define (since it's actually possible to do a send transaction that actually sends multiple tokens, which is not the meaning here)